File: 9-palindrome-number/palindrome-number.dart
class Solution {
bool isPalindrome(int x) {
  if (x < 0) {
    return false;
  }

  int reminder = x % 10;
  int result = 0;
  int temp = x;

  while (temp != 0) {
    reminder = temp % 10;
    result = (result * 10) + reminder;
    temp = temp ~/ 10;
  }

  if (x != result) {
    return false;
  }
  return true;
}

  
}

File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int r, sum = 0;
        int n = x; // Initialize n with the value of x

        if (x < 0) {
            return false;
        } else {
            while (n > 0) {
                r = n % 10;
                sum = (sum * 10) + r;
                n = n / 10;
            }

            return sum == x;
        }

    }
}
